plugins/logs/plugin.go

```diff
@@ -675,29 +680,20 @@ func (p *Plugin) maskEvent(ctx context.Context, txn storage.Transaction, event *
 
 			pq, err := r.PrepareForEval(context.Background())
 			if err != nil {
-				return err
+				return nil, err
 			}
 
 			p.mask = &pq
 		}
 
-		return nil
-	}()
+		rs, err := p.mask.Eval(
+			ctx,
+			rego.EvalParsedInput(input),
+			rego.EvalTransaction(txn),
+		)
 
-	if err != nil {
-		return err
-	}
-
-	input, err := event.AST()
-	if err != nil {
-		return err
-	}
-
-	rs, err := p.mask.Eval(
-		ctx,
-		rego.EvalParsedInput(input),
-		rego.EvalTransaction(txn),
-	)
+		return rs, err
```

Comment:
Instead of evaluating the mask policy while holding the mutex, assign the value of `p.mask` to a local variable and then release the mutex. We don't want to hold the mutex while evaluating the mask policy.